source/SH3/graphics/msbmp.cpp

```diff
@@ -0,0 +1,100 @@
+/** @file
+ *
+ *  @copyright 2016  Palm Studios
+ *
+ *  @date 6-2-17
+ *
+ *  @author Jesse Buhagiar
+ */
+
+#include "SH3/graphics/msbmp.hpp"
+#include "SH3/system/log.hpp"
+
+#include <fstream>
+
+using namespace sh3_graphics;
+
+msbmp::load_result msbmp::Load(const std::string& path)
+{
+    std::ifstream   file;
+    bmp_file_header fheader; // File information header
+    bmp_info_header iheader;
+    load_result res;
+
+    res.set_error(result::BAD_LOAD);
+    file.open(path, std::ios::binary);
+
+    if(!file.is_open())
+    {
+        // If this happens we're in a bit of trouble, because we can't load the default texture..... We could crash here.
+        Log(LogLevel::ERROR, "msbmp::Load( ): Unable to open a handle to %s!", path.c_str());
+        return res;
+    }
+
+    file.read((char*)&fheader, sizeof(fheader)); // Read in the file header
```

Comment:
Ahh okay, I remember why I used the C style cast now, I can't compile using `palette.data()`